src/utils/helpers.tsx

```diff
@@ -141,6 +141,10 @@ export const isEqual = (a: unknown, b: unknown): boolean => {
         return true;
     }
 
+    if (typeof a === 'number' && typeof b === 'number' && isNaN(a) && isNaN(b)) {
```

Comment:
According to [lodash documentation](https://lodash.com/docs/4.17.15#isEqual), `isEqual(NaN, NaN)` should return `true`:

![image](https://github.com/Telefonica/mistica-web/assets/25785151/560caf71-8f90-4b80-8eba-38ada6a65bf4)

But TS returns false when comparing NaN with itself:

![image](https://github.com/Telefonica/mistica-web/assets/25785151/03be05a5-b833-40ee-8325-214776d93f41)

Reply:
not TS, this is JS and it is expected to be false:

> NaN compares unequal (via [==](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality), [!=](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Inequality), [===](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality), and [!==](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_inequality)) to any other value — including to another NaN value.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN

Reply:
Yes, It's expected in JS, but we don't want that for this helper

Reply:
please include a unit test

Reply:
done